templates/goshared/msg.go

```diff
@@ -43,6 +64,21 @@ func (m {{ (msgTyp .).Pointer }}) Validate() error {
 
 {{ if needs . "uuid" }}{{ template "uuid" . }}{{ end }}
 
+{{ cmt (multierrname .) " is an error wrapping multiple validation errors returned by " (msgTyp .) ".Validate(true) if the designated constraints aren't met." -}}
+type {{ multierrname . }} []error
+
+// Error returns a concatenation of all the error messages it wraps.
+func (m {{ multierrname . }}) Error() string {
+	var msgs []string
+	for _, err := range m {
+		msgs = append(msgs, err.Error())
+	}
+	return strings.Join(msgs, "; ")
+}
+
+// AllErrors returns a list of validation violation errors.
+func (m {{ multierrname . }}) AllErrors() []error { return m }
```

Comment:
For nested scenarios, I wonder if it's worth adding a message that returns all errors recursively as well to completely flatten a tree of errors?  If you feel like that complexity might be too much to implement here I'm more than happy to agree that it should be implemented in clients b/c this is already a great step.

Reply:
Thing is, such a func is quite complex, and yet basically 100% operates on interfaces (doesn't need to be specialized per-error-type), so IMO it doesn't need to be a method, and potentially it could be put as a "helper" func in an external package, e.g. a subpackage of PGV for being imported separately. Still, I think it would make quite a lot of sense to add such a func (or even method if really insisted), but I'd personally vote for splitting it to a separate PR. And @brandoncole  if you're really interested in this topic, maybe you'd like to try your hand at writing such a PR yourself? 🙂